### PR TITLE
AEA-1468 Feature/misc

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR addresses multiple issues:
- improves contributor guide
- removes global ignore of `"unsubscriptable-object"` for pylint and applies it for protocols' messages only
- addresses https://github.com/fetchai/agents-aea/issues/2392 and makes try decorator behaviour configurable as well as raises, rather than logs exceptions so the try decorator can then handle them as configured.

## Fixes

-https://github.com/fetchai/agents-aea/issues/2392

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/helpers/base.py [hunk 1]
```diff
@@ -312,6 +312,8 @@ def wrapper(*args: Any, **kwargs: Any) -> Callable:
             try:
                 return fn(*args, **kwargs)
             except Exception as e:  # pylint: disable=broad-except  # pragma: no cover  # generic code
+                if len(args) > 0 and getattr(args[0], "raise_on_try", False):
+                    raise e
```
Comment (DavidMinarsch) [inline aea/helpers/base.py:316]: Say in a ledger api, I can now do `ledger_api.raise_on_try = True` and then use it programmatically without exceptions being caught early.

File: plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py [hunk 2]
```diff
@@ -1163,6 +1189,10 @@ def _try_get_transaction_receipt(self, tx_digest: str) -> Optional[JSONLike]:
         response = requests.get(url=url)
         if response.status_code == 200:
             result = response.json()
+        else:  # pragma: nocover
+            raise ValueError(
+                "Cannot get transaction receipt: {}".format(response.json())
+            )
         return result
```
Comment (marcofavorito) [inline plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py:1196]:
```suggestion
            return response.json()
        raise ValueError(  # pragma: nocover
            "Cannot get transaction receipt: {}".format(response.json())
        )

```

**Review by jrriehl — APPROVED** [review]
Looks good! Better error handling / more informative error messages in ledger apis was very much needed.

File: plugins/aea-ledger-cosmos/aea_ledger_cosmos/cosmos.py [hunk 3]
```diff
@@ -661,6 +667,8 @@ def _try_get_state(  # pylint: disable=unused-argument
         response = requests.get(url=url)
         if response.status_code == 200:
             result = response.json()
+        else:  # pragma: nocover
+            raise ValueError("Cannot get state: {}".format(response.json()))
         return result
```
Comment (marcofavorito) [inline plugins/aea-ledger-cosmos/aea_ledger_cosmos/cosmos.py:672]:
```suggestion
        if response.status_code == 200:
            return response.json()
        raise ValueError("Cannot get state: {}".format(response.json()))  # pragma: nocover
```

File: plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py (hunk 2 above)
Comment (marcofavorito) [inline plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py:1196]:
```suggestion
            return response.json()
        raise ValueError(  # pragma: nocover
            "Cannot get transaction receipt: {}".format(response.json())
        )
        return result
```

File: plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py [hunk 4]
```diff
@@ -1333,9 +1363,10 @@ def _try_create_faucet_claim(
 
         uid = None
         if response.status_code == 200:
-            data = response.json()
-            uid = data["uid"]
-
+            try:
+                uid = response.json()["uid"]
```
Comment (marcofavorito) [inline plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py:1367]: is `KeyError` the only exception that can be raised here? What about `JSONDecodeError`: https://docs.python-requests.org/en/master/user/quickstart/#json-response-content

**Review by marcofavorito — APPROVED** [review]
LGTM :+1:

**Pull request merged**